src/librustc_codegen_llvm/builder.rs

```diff
@@ -1069,80 +1148,101 @@ impl Builder<'a, 'll, 'tcx> {
         ret.expect("LLVM does not have support for catchswitch")
     }
 
-    pub fn add_handler(&self, catch_switch: &'ll Value, handler: &'ll BasicBlock) {
+    fn add_handler(&mut self, catch_switch: &'ll Value, handler: &'ll BasicBlock) {
         unsafe {
             llvm::LLVMRustAddHandler(catch_switch, handler);
         }
     }
 
-    pub fn set_personality_fn(&self, personality: &'ll Value) {
+    fn set_personality_fn(&mut self, personality: &'ll Value) {
         unsafe {
             llvm::LLVMSetPersonalityFn(self.llfn(), personality);
         }
     }
 
     // Atomic Operations
-    pub fn atomic_cmpxchg(
-        &self,
+    fn atomic_cmpxchg(
+        &mut self,
         dst: &'ll Value,
         cmp: &'ll Value,
         src: &'ll Value,
-        order: AtomicOrdering,
-        failure_order: AtomicOrdering,
-        weak: llvm::Bool,
+        order: rustc_codegen_ssa::common::AtomicOrdering,
+        failure_order: rustc_codegen_ssa::common::AtomicOrdering,
+        weak: bool,
     ) -> &'ll Value {
-        unsafe {
-            llvm::LLVMRustBuildAtomicCmpXchg(self.llbuilder, dst, cmp, src,
-                                             order, failure_order, weak)
+        let weak = if weak { llvm::True } else { llvm::False };
+        unsafe {
+            llvm::LLVMRustBuildAtomicCmpXchg(
+                self.llbuilder,
+                dst,
+                cmp,
+                src,
+                AtomicOrdering::from_generic(order),
+                AtomicOrdering::from_generic(failure_order),
```

Comment:
This makes me think `From` should be implemented instead. Then, the LLVM-specific `AtomicOrdering` do not need to be imported, removing `rustc_codegen_ssa::common::` noise in the signature *and* we can just do `order.into()` and whatnot.